### PR TITLE
test: migrated content-length.test.js from tap to node:test

### PR DESCRIPTION
**Pull request by Tony133**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

Proposal:

  -  migrated `content-length.test.js` from `tap` to `node:test` 🔥

### CODE REVIEW

**Review by gurgunday — COMMENTED** [review]
CI is red...

**Comment by Tony133** [comment]
> CI is red..

Yes, I just don't know exactly why, the updated test with `node:test` seems correct  🤔
test-ok

File: test/content-length.test.js [hunk 1]
```diff
@@ -180,10 +174,9 @@ test('#2543 - wrong content-length with errorHandler', t => {
     method: 'GET',
     path: '/'
   })
-    .then(res => {
-      t.equal(res.statusCode, 500)
-      t.equal(res.headers['content-length'], '' + res.rawPayload.length)
-      t.same(JSON.parse(res.payload), { message: 'longer than 2 bytes' })
-      t.end()
+    .then(response => {
```
Comment (Uzlopak) [inline test/content-length.test.js:177]: let me guess, inject would throw?

Comment (Tony133) [reply test/content-length.test.js]: made another change see commit here: https://github.com/fastify/fastify/pull/5878/commits/febdf6f169769dea754dcf649d05ebdf17fd355d

but it still fails

File: test/content-length.test.js [hunk 2]
```diff
@@ -15,26 +14,25 @@ test('default 413 with bodyLimit option', t => {
...
-    t.equal(res.statusCode, 413)
-    t.equal(res.headers['content-type'], 'application/json; charset=utf-8')
-    t.same(JSON.parse(res.payload), {
-      error: 'Payload Too Large',
-      code: 'FST_ERR_CTP_BODY_TOO_LARGE',
-      message: 'Request body is too large',
-      statusCode: 413
-    })
+  })
+  t.assert.ifError(response.err)
```
Comment (Eomm) [inline test/content-length.test.js:24]: `response.err` does not exist - we need to update the plan and remove this line

File: test/content-length.test.js [hunk 3]
```diff
@@ -52,20 +50,19 @@ test('default 400 with wrong content-length', t => {
...
-    t.equal(res.statusCode, 400)
-    t.equal(res.headers['content-type'], 'application/json; charset=utf-8')
-    t.same(JSON.parse(res.payload), {
-      error: 'Bad Request',
-      code: 'FST_ERR_CTP_INVALID_CONTENT_LENGTH',
-      message: 'Request body size did not match Content-Length',
-      statusCode: 400
-    })
+  })
+  t.assert.ifError(response.err)
```
Comment (Eomm) [inline test/content-length.test.js:54]: `response.err` does not exist - we need to update the plan and remove this line

File: test/content-length.test.js [hunk 4]
```diff
@@ -83,26 +80,25 @@ test('custom 413 with bodyLimit option', t => {
...
-    t.equal(res.statusCode, 413)
-    t.equal(res.headers['content-type'], 'application/json; charset=utf-8')
-    t.same(JSON.parse(res.payload), {
-      error: 'Payload Too Large',
-      code: 'FST_ERR_CTP_BODY_TOO_LARGE',
-      message: 'Request body is too large',
-      statusCode: 413
-    })
+  })
+  t.assert.ifError(response.err)
```
Comment (Eomm) [inline test/content-length.test.js:90]: `response.err` does not exist - we need to update the plan and remove this line

File: test/content-length.test.js [hunk 5]
```diff
@@ -127,20 +123,19 @@ test('custom 400 with wrong content-length', t => {
...
-    t.equal(res.statusCode, 400)
-    t.equal(res.headers['content-type'], 'application/json; charset=utf-8')
-    t.same(JSON.parse(res.payload), {
-      error: 'Bad Request',
-      code: 'FST_ERR_CTP_INVALID_CONTENT_LENGTH',
-      message: 'Request body size did not match Content-Length',
-      statusCode: 400
-    })
+  })
+  t.assert.ifError(response.err)
```
Comment (Eomm) [inline test/content-length.test.js:127]: `response.err` does not exist - we need to update the plan and remove this line

**Review by Eomm — COMMENTED** [review]
The error seems a flaky one

```
FAILED: child test left in queue: onRequest hook should support encapsulation / 2
```

maybe we forget to call a done in the right time?

File: test/content-length.test.js (hunk 2 above)
Comment (Tony133) [reply test/content-length.test.js]: I have updated the plan see commit here: [b5f4c7c](https://github.com/fastify/fastify/pull/5878/commits/b5f4c7c7b0ecbf5ae59c3f231ff86fa466415030), but it still fails

**Review by gurgunday — COMMENTED** [review]
windows coverage is failing :(

**Comment by Eomm** [comment]
> windows coverage is failing :(

the issue seems related to this https://github.com/MoLow/reporters/pull/131 and not to this PR - other PRs are failing with the same error

**Review by Eomm — APPROVED** [review]

**Pull request merged**